Adding form to update quantity and some additional css for cart

Tman22 opened the pull request:
A few additional things added are:
Css for buttons on cart page.
Cart icon got a counter for total.
Shoulda matcher was almost complete,
but couldn't get gem to configure correctly.

Tman22 committed:
Add form for cart to update quantity

A few additional things added are:
Css for buttons on cart page.
Cart icon got a counter for total.
Shoulda matcher was almost complete,
but couldn't get gem to configure correctly.

Tman22 commented:
I had some issues with shoulda matcher. I might have found the right version before I pushed this but need to write the correct syntax for testing.

Tman22 commented:
Also the css for the cart icon in the _navbar has a counter that is being displayed below the cart. I'd love to get some help if anyone can look at this and help me fix this css.

ToniRib commented on app/models/cart.rb:
```diff
@@ -10,6 +10,10 @@ def add_item(item_id)
   end
 
+  def update_quantity(item_id, new_quantity)
+    contents[item_id] = new_quantity.to_i
```
Do we need the to_s here, or is it already coming in as a string?

Tman22 replied:
new_quantity comes in as a string so I changed it to a integer because contents[item_id] is set as a integer. Currently I'm working on how we want to arrange the routes. Right now I redirect_to cart_path (index). I wasn't sure if I should try and render :index instead? The issue here is I send the information through the url which wouldn't go with the user story (which requests current_path to be '/cart' after updating). If we would like to render instead I will look into reconfiguring the routes and adding hidden_fields.

ToniRib commented on app/models/cart.rb:
```diff
@@ -20,6 +24,10 @@ def item_quantities
   end
 
+  def count
+    contents.values.reduce(:+)
```
If I knew how to emoji this, I would give you a thumbs up.

ToniRib commented on test/integration/guest_can_change_quanity_in_cart_test.rb:
```diff
@@ -0,0 +1,43 @@
+
+class GuestCanChangeQuanityInCartTest < ActionDispatch::IntegrationTest
+  test "guest visits cart and adds more items" do
+    create_and_add_item_to_cart
```
I forget, are we using this method in more than one test? If so, maybe we can move it into the test helper.

Tman22 replied:
Yes, I was going to extract this before I merge with master. We use in the guest can remove from cart test.

ToniRib commented on test/integration/guest_can_change_quanity_in_cart_test.rb:
```diff
@@ -0,0 +1,43 @@
+  #     And when I increase the quantity
+  #     Then my current page should be '/cart'
+  #     And that item's quantity should reflect the increase
+  #     And the subtotal for that item should increase
```
I just realized that we are not showing subtotals, so we need to add that to our cart page (within the item partial I assume).

Tman22 committed:
Add subtotal to cart.

Branch has subtotal and some css modifications to the navbar
which brings the count of the cart next to the cart icon.

ToniRib committed:
User can edit item quantity in cart

Item quantity in the cart is an editable field linked to a form
and an update button so the quantity can be changed. The total number
of items in the cart is also displayed next to the shopping cart icon.
Note this branch is called 'model test' but that will actually be
worked on on a different branch.

ToniRib committed:
Merge branch 'master' of https://github.com/martensonbj/little_shop into model_testing

ToniRib commented:
Everything looks good, all tests are passing. Merging in the new functionality.
